src/linear_solver_logs/solve_log_ma.jl

```diff
@@ -183,17 +184,18 @@ function log_update!(
     if !log.true_res && iter > 0
         # Compute the current residual to second power to align with theory
         # Check if it is one dimensional or block sampling method
-        res::Float64 = eltype(samp[1]) <: Int64 || size(samp[1],2) != 1 ? 
-            log.resid_norm(samp[3])^2 : log.resid_norm(dot(samp[1], x) - samp[2])^2 
+        res::Float64 = log.dist_info.scaling * (eltype(samp[1]) <: Int64 || size(samp[1],2) != 1 ? 
+                                                log.resid_norm(samp[3])^2 : log.resid_norm(dot(samp[1], x) - samp[2])^2) 
     else 
         res = log.resid_norm(A * x - b)^2 
     end
+  
     # Check if MA is in lambda1 or lambda2 regime
     if ma_info.flag
         update_ma!(log, res, ma_info.lambda2, iter)
     else
         #Check if we can switch between lambda1 and lambda2 regime
-        if res < ma_info.res_window[ma_info.idx]
+        if iter == 0 || res <= ma_info.res_window[ma_info.idx] 
```

Comment:
This line needs more explanation

Reply:
I have clarified that this lean is checking for a monotonic decrease in the sketched residuals